src/vs/workbench/browser/parts/views/viewPaneContainer.ts

```diff
@@ -982,6 +982,8 @@ export class ViewPaneContainer extends Component implements IViewPaneContainer {
 				this.updateViewHeaders();
 			}
 		});
+
+		this._register(this.viewContainerModel.onDidChangeActiveViewDescriptors(() => this._onTitleAreaUpdate.fire()));
```

Comment:
May I know why this is it needed?

Reply:
A view might have a `when` property that causes it to become applicable later (the NPM view does this). The view menu in the `...` menu needs to respond to this update. Both lines that you commented on fix this.

This is what I'm talking about:
![recording (8)](https://user-images.githubusercontent.com/38270282/87405480-cdc6ba80-c5bf-11ea-8f82-9fab2c878e08.gif)

Reply:
Ok. Makes sense.